### PR TITLE
Fix subscribe request aggregation in `SharedWorker`.

### PR DESCRIPTION
**Pull request by parfeon**
fix(shared-worker): fix presence leave request preparation

Fix issue because of which leave request modified wrong URL path component with actual channels.

fix(shared-worker): fix channels / groups list modification issues

Fix issue because of which removed channels / groups didn't cancel previous subscribe request to re-subscribe with new set of channels / groups.

fix(shared-worker): fix requests aggregation issues

Fix issue because of which suitable active PubNub clients subscription not has been used for aggregation and caused additional connections or wrong set of channels / groups.

refactor(subscription): process messages only for subscribed

Pre-process entries from subscribe response to filter out updates which has been received for channels and groups which are not part of subscription loop (subscription aggregation in shared worker).

refactor(package): add `types` entry for Node.js integration

Point to the built-in types definition file when package used with `npm` / `yarn`.

### CODE REVIEW

File: src/core/endpoints/subscribe.ts [hunk 1]
```diff
@@ -629,57 +629,65 @@ export class BaseSubscribeRequest extends AbstractRequest<Subscription.Subscript
       );
     }
 
-    const events: Subscription.SubscriptionResponse['messages'] = serviceResponse.m.map((envelope) => {
-      let { e: eventType } = envelope;
+    const events: Subscription.SubscriptionResponse['messages'] = serviceResponse.m
+      .filter((envelope) => {
+        const subscribable = envelope.b === undefined ? envelope.c : envelope.b;
```
Comment (mohitpubnub) [inline src/core/endpoints/subscribe.ts:634]: 👍🏻

File: src/transport/subscription-worker/subscription-worker.ts [hunk 2]
```diff
@@ -589,7 +627,21 @@ self.onconnect = (event) => {
...
         if (data.request.path.startsWith('/v2/subscribe')) {
           updateClientStateIfRequired(data);
-          handleSendSubscribeRequestEvent(data);
+
+          const client = pubNubClients[data.clientIdentifier];
+          if (client) {
+            const timerIdentifier = `${client.userId}-${client.subscriptionKey}`;
+
+            // Check whether we need to start new aggregation timer or not.
+            if (!aggregationTimers.has(timerIdentifier)) {
```
Comment (mohitpubnub) [inline src/transport/subscription-worker/subscription-worker.ts:636]: 👍🏻

**Review by mohitpubnub — APPROVED** [review]
LGTM!

**Comment by parfeon** [comment]
@pubnub-release-bot release

**Pull request merged**